internal/spdxlicense/license.go

```diff
@@ -1,12 +1,38 @@
 package spdxlicense
 
 import (
+	"regexp"
 	"strings"
 )
 
+// https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/#license-short-name
+// If a license generated in license_list.go is not found when looking up by ID,
+// then the ID function will use these regular expressions to help resolve cases where
+// x.0.0 and x are supplied as version numbers. For SPDX compatibility, versions with trailing
+// dot-zeroes are considered to be equivalent to versions without (e.g., “2.0.0” is considered equal to “2.0” and “2”).
+// EX: gpl-2+ ---> GPL-2.0+
+// EX: gpl-2.0.0-only ---> GPL-2.0-only
+// See the debian link for more details on the spdx license differences
+var (
+	zero   = regexp.MustCompile(`^((.*).0)(.*)$`)
+	noZero = regexp.MustCompile(`^(.*-)([1-9])(.*)`)
```

Comment:
I don't think it's a _huge_ problem for this at all, but the first regex will match things like `GPL-2.0.1`, later becoming `GPL-2.1` maybe the first regex should be `^(.*\.0)\.0(.*)$` and do `idBytes = zero.ReplaceAll([]byte(lowerID), []byte("${1}${2}"))`? and the second would match `GPL-2.0.1-stuff` and later become `GPL-2.0.1.0-stuff`, maybe use `^(.*\-\d)([^.1-9]+)$`? and later `noZero.ReplaceAll([]byte(lowerID), []byte("${1}.0${2}"))`?

Reply:
Nice! I've update these to be more exclusive. I think the only spot where it's a little funky now is the `noZero` case.

`gpl-2` is not matched on anymore which causes the ReplaceAll to not fire just leaving us with `gpl-2` and no trailing `.0`